### PR TITLE
Add WG Serving

### PR DESCRIPTION
**Pull request by animeshsingh**
WG Serving Proposal

|       Person       	|               WG              	| Response 	|
|:------------------:	|:-----------------------------:	|:--------:	|
| @animeshsingh      	| KFServing, Pipelines          	|          	|
| @cliveseldon       	| KFServing                     	|          	|
| @yuzisun           	| KFServing                     	|          	|
| @ellistarn         	| KFServing                     	|          	|
| @neuromage         	| Pipelines/Metadata            	|          	|
| @paveldournov      	| Pipelines/Metadata            	|          	|
| @elikatsis         	| Pipelines                     	|          	|
| @vpavlin           	| Control plane                 	|          	|
| @yanniszark        	| Control Plane/common services 	|          	|
| @Jeffwan           	| Control Plane                 	|       LGTM   	|
| @krishnadurai      	| Common Services               	|          	|
| @terrytangyuan     	| Training                      	|      LGTM    	|
| @gaocegege         	| Training                      	|      	|
| @andreyvelich      	| Training                      	|          	|
| @johnugeorge       	| Training                      	|     LGTM     	|
| @David Aronchick         	| Metadata                      	|          	|
| @StefanoFioravanzo 	| Metadata                      	|          	|
| @jbottum           	| PM                            	|          	|
| @elviraux          	| PM                            	|          	|
| @kimwnasptd        	| Notebooks                     	|          	|
| @krazyhaas         	|                               	|          	|

### CODE REVIEW

File: wgs/wg-serving/wg-charter-serving.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+#### Cross-cutting and Externally Facing Processes
+
+- Coordinating with WG Pipeline to make sure models can be deployed and rolled out  with pipelines
+- Coordinating with Kubeflow release teams to ensure that the serving features can be released properly (e.g. integration and testing Istio versions supported by Kubeflow community, providing kustomization manifests etc.)
+- Coordinating with various cloud providers to ensure KFServing can be deployed and operated at their end.
+
+### Out of scope
+
+- APIs used for running distributed training, hyper parameter optimization, neural architecture search etc.
```
Comment (ellistarn) [inline wgs/wg-serving/wg-charter-serving.md:45]: Add feature stores.

File: wgs/wg-serving/wg-charter-serving.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+- Monitoring: Are the endpoints healthy? What is the performance profile and request trace? 
+- Rollouts: Is this rollout safe? How do I roll back? Can I test a change without swapping traffic?
+- Protocol Standards: How do I make a prediction? GRPC? HTTP? Kafka?
+- Frameworks: How do I serve on Tensorflow? XGBoost? Scikit Learn? Pytorch? Custom Code? 
+- Features: How do I explain the predictions? What about detecting outliers and skew? Bias detection? Adversarial Detection
+- Batch: How do I handle batch predictions? 
+- Data Plane: How do I leverage standardized Data Plane protocol so that I can move my model across MLServing platforms?
+- Custom Processors: How do I wire up custom pre and post processing
+
+### In scope
```
Comment (ellistarn) [inline wgs/wg-serving/wg-charter-serving.md:19]: Add feature store integration

File: wgs/wg-serving/wg-charter-serving.md (hunk 1 above)
Comment (animeshsingh) [reply wgs/wg-serving/wg-charter-serving.md]: done

File: wgs/wg-serving/wg-charter-serving.md (hunk 2 above)
Comment (animeshsingh) [reply wgs/wg-serving/wg-charter-serving.md:19]: done

**Comment by animeshsingh** [comment]
cc @jlewi @jbottum

**Comment by jlewi** [comment]
Thanks @animeshsingh ; can you please publicize this and try to get other leads to weigh in?

On my end LGTM.

Given you have 7 leads; Did you consider trying to divy up responsibilities a bit more clearly? i.e. instead of having everyone where both hats of manager and tech leads; have more folks pick one or the other?

**Comment by animeshsingh** [comment]
thanks @jlew . Made quick edits. Consistent with other WGs to begin with, and we can evolve over a period of time based on how much and org and individual wants to remain involved (set direction for project, or get involved in day to day dev, maintenance and integration work scenarios).

cc @ellistarn @cliveseldon @yuzisun @Iamlovingit @rakelkar @deadeyegoodwin

File: wgs/wg-serving/wg-charter-serving.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+- Multiple clients: kubectl, Python SDK, Kubeflow Pipelines SDK
+- Data Plane: Standardized Data Plane V2 protocol for prediction/explainability
+- Multi Model Serving: Serving multiple models per custom KFService instance
+- Ensembles: For composable inference graphs and support for chaining multiple models together
+- Multiple protocols: streaming, gRPC Support for Model Servers
+- Feature Store Integration: Integrate with feature store like Feast
+
+#### Cross-cutting and Externally Facing Processes
+
+- Kubeflow Pipelines Integration: Coordinating with WG Pipeline to make sure models can be deployed and rolled out  with pipelines
```
Comment (Jeffwan) [inline wgs/wg-serving/wg-charter-serving.md:39]: minor: extra space between `out` and `with`

File: wgs/wg-serving/wg-charter-serving.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
+# WG Serving Charter
+
+This charter adheres to the conventions, roles and organization management outlined in [wg-governance].
```
Comment (Jeffwan) [inline wgs/wg-serving/wg-charter-serving.md:3]: Trying to understand the subprojects of this WG. Currently, I think it's kfserving, what's the status of seldon? If we have more serving projects in the future. For example, bentoML wants to integrate more with KFServing or other components. they are all under this WG, right?

Comment (animeshsingh) [reply wgs/wg-serving/wg-charter-serving.md:3]: Seldon is virtually part of every dicussion we have, courtesy @cliveseldon being there. We are striving to do tighter collaboration with Seldon as part of WG, and though not listed explicitly here, we will be adding more on it soon.

I will list both Seldon and BentoML currently under the cross-cutting concerns here as well.

File: wgs/wg-serving/wg-charter-serving.md (hunk 3 above)
Comment (animeshsingh) [reply wgs/wg-serving/wg-charter-serving.md]: done

File: wgs/wg-serving/wg-charter-serving.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+- Ensembles: For composable inference graphs and support for chaining multiple models together
+- Multiple protocols: streaming, gRPC Support for Model Servers
+- Feature Store Integration: Integrate with feature store like Feast
+
+#### Cross-cutting and Externally Facing Processes
+
+- Kubeflow Pipelines Integration: Coordinating with WG Pipeline to make sure models can be deployed and rolled out with pipelines
+- Kubeflow Integration: Coordinating with Kubeflow release teams to ensure that the serving features can be released properly (e.g. integration and testing Istio versions supported by Kubeflow community, providing kustomization manifests etc.)
+- Cloud Providers: Coordinating with various cloud providers to ensure KFServing can be deployed and operated at their end.
+- Model Serving Platforms: Explore integration oppurtinities with other model serving platforms like Seldon, BentoML etc.
```
Comment (andreyvelich) [inline wgs/wg-serving/wg-charter-serving.md:42]:
```suggestion
- Model Serving Platforms: Explore integration opportunities with other model serving platforms like Seldon, BentoML etc.
```

File: wgs/wg-serving/wg-charter-serving.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+- Features: How do I explain the predictions? What about detecting outliers and skew? Bias detection? Adversarial Detection
+- Batch: How do I handle batch predictions? 
+- Data Plane: How do I leverage standardized Data Plane protocol so that I can move my model across MLServing platforms?
+- Custom Processors: How do I wire up custom pre and post processing
+
+### In scope
+
+#### Code, Binaries and Services
+
+- APIs: For deploying models with multiple runtimes pre-integrated (TFServing, Nvdia Triton (GPU optimization), ONNX Runtime, SKLearn, PyTorch, XGBoost, Custom models)
```
Comment (Tomcli) [inline wgs/wg-serving/wg-charter-serving.md:23]:
```suggestion
- APIs: For deploying models with multiple runtimes pre-integrated (TFServing, Nvidia Triton (GPU optimization), ONNX Runtime, SKLearn, PyTorch, XGBoost, Custom models)
```

File: wgs/wg-serving/wg-charter-serving.md (hunk 5 above)
Comment (animeshsingh) [reply wgs/wg-serving/wg-charter-serving.md]: done

**Pull request merged**